### PR TITLE
Fix for #523

### PR DESCRIPTION
**Pull request by sverkera**
This issue is caused by the github lib on these occasions will catch an IOException and wrap it in a java.lang.Error. This plugin has a catch-all in the thread for Exceptions, but that does not catch an Error.

I've changed that to catch Throwable, which will catch both Error and Exception. I've also added some catching of Error higher up where it catches IOException.

Edit by @samrocketman: Fix for #523

### CODE REVIEW

**Comment by anauckunas** [comment]
Can someone please merge this

**Comment by bjoernhaeuser** [comment]
can you link the code which wraps the exception in an error? That alreadys sound off and should be changed - IMHO.

File: src/main/java/org/jenkinsci/plugins/ghprb/GhprbRootAction.java [hunk 1]
```diff
@@ -250,13 +250,13 @@ private void triggerPr(final GhprbTrigger trigger,
             public void run() {
                 try {
                     trigger.handlePR(pr);
-                } catch (Exception e) {
+                } catch (Throwable th) {
```
Comment (samrocketman) [inline src/main/java/org/jenkinsci/plugins/ghprb/GhprbRootAction.java]: In general, I've been told catching Throwable is a bad idea.  Can a more seasoned Jenkins dev review this PR?

cc @jenkinsci/code-reviewers

Comment (samrocketman) [reply src/main/java/org/jenkinsci/plugins/ghprb/GhprbRootAction.java]: If the intention here is to catch `Exception` and `Error`, then check for those types and if it's not either of those types then rethrow throwable.  I'm just concerned that we're swallowing all throwable.

Comment (samrocketman) [reply src/main/java/org/jenkinsci/plugins/ghprb/GhprbRootAction.java]: Also a quote from the `java.lang.Error` javadoc:

> An Error is a subclass of Throwable that indicates serious problems that a reasonable application should not try to catch.

I need an adult 🆘

Comment (oleg-nenashev) [reply src/main/java/org/jenkinsci/plugins/ghprb/GhprbRootAction.java]: It's a bad style for sure. But, you know, throwing it from the code is a bad idea as well :(

**Comment by samrocketman** [comment]
@bjoernhaeuser I'm headed to bed now.  I trust @oleg-nenashev judgement on this one.  When the PR finishes verifying do you mind merging?  I'll get to it tomorrow if it's not merged so not a big deal if you don't have time.

File: src/main/java/org/jenkinsci/plugins/ghprb/GhprbRootAction.java (hunk 1 above)
Comment (jglick) [reply src/main/java/org/jenkinsci/plugins/ghprb/GhprbRootAction.java]: In practice the main things it is problematic to catch are `OutOfMemoryError` (generally no practical recovery, so does not make much difference either way); and `ThreadDeath` (you need to be sure to rethrow quickly at least).

Comment (samrocketman) [reply src/main/java/org/jenkinsci/plugins/ghprb/GhprbRootAction.java]: I'll open up a followup PR @jglick.  I guess I merged too quickly.

**Pull request merged**